clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy

```diff
@@ -110,8 +113,9 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
       } catch (AmazonServiceException ignore) {
       }
 
-      def securityGroups = regionScopedProvider.securityGroupService.
-              getSecurityGroupIds(description.securityGroups, description.vpcId).values()
+      def securityGroupNamesToIds = regionScopedProvider.securityGroupService.getSecurityGroupIds(description.securityGroups, description.vpcId)
+      def securityGroups = securityGroupNamesToIds.values()
+      updateApplicationSecurityGroupIngressRules(securityGroupNamesToIds, listeners, regionScopedProvider)
```

Comment:
We also want to include the existing security group rules, in addition to whatever the listener is providing: If the application security group has additional rules that aren't related to the load balancer, this code will unset those rules.